europeana-image-harvester/index.html

```diff
@@ -0,0 +1,146 @@
+            .then(response => {
+              if (response.valid) {
+                document.getElementById('message').innerText = 'Processing';
+                clearFields();
+                populateFields(response.json, response.recordId);
+              } else {
+                document.getElementById('message').innerText = response.error;
+                document.getElementById('message').style.color = '#d9453f';
+              }
+              document.getElementById('message').style.display = 'block';
+            });
+        }
+
+        // get the image and attribution
+        async function harvestImageAndAttribution (recordUrl) {
+          const recordId = recordUrl.match('\/[0-9]+\/[a-zA-Z0-9_]+$')[0];
+          const apiKey = extension.parameters.installation.apiKey;
+
+          const response = await fetch(`https://api.europeana.eu/api/v2/record${recordId}.json?wskey=${apiKey}`);
+          const responseJson = await response.json();
+          if (response.ok) {
+            return { valid: true, json: responseJson, recordId: recordId };
+          } else {
+            return { valid: false, error: responseJson.error ? responseJson.error : 'unknown' };
+          }
+        }
+
+        // populate fields
+        async function populateFields(response, recordId) {
+          const image = response.object.aggregations[0].edmIsShownBy;
+          if (!image) {
+            document.getElementById('message').innerText = 'No image found';
+            document.getElementById('message').style.color = '#d9453f';
+            return;
+          }
+          
+          // look up image info, attribution (title, provider, creator) and rights statement
+          const providerAggregation = response.object.aggregations[0];
+          const resource = providerAggregation.webResources.find( r => {
+            return r.about === image;
+          });
+
+          const proxies = response.object.proxies;
+          const providerProxy = proxies.find( p => {
+            return p.europeanaProxy === false;
+          });
+
+          const attribution = [languageValue(providerProxy.dcTitle),
+                               languageValue(providerProxy.dcCreator),
+                               languageValue(providerAggregation.edmDataProvider)];
+
+          extension.entry.fields.identifier.setValue(recordId);
+          extension.entry.fields.attribution.setValue(attribution.filter(Boolean).join(', '));
+          extension.entry.fields.rightsStatement.setValue(languageValue(providerAggregation.edmRights));
+
+          // create an asset
+          const asset = {
+            fields: {
+              title: {},
+              file: {}
+            }
+          }
+          asset.fields.title['en-GB'] = languageValue(providerProxy.dcTitle) ? languageValue(providerProxy.dcTitle) : recordId;
+          asset.fields.file['en-GB'] = {
+            contentType: resource.ebucoreHasMimeType,
```

Comment:
What happens if this is blank in the API response?

Reply:
There is a fallback now to 'application/octet-stream', which seems the most default one. Does that make sense?